### PR TITLE
Instance sharing error messages

### PR DESCRIPTION
**Pull request by EpsilonTal**
Modifying the instance sharing error messages.

### CODE REVIEW

File: pkg/util/errors.go [hunk 1]
```diff
@@ -258,7 +257,7 @@ func HandleInstanceSharingError(err error, entityName string) error {
 	case ErrCatalogUsesReservedPlanName:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("The plan name you used for registration \"%s\" is a reserved name; you must choose a different name.", entityName),
+			Description: fmt.Sprintf("The plan name \"%s\" is a reserved name; you must choose a different plan name.", entityName),
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:260]: Catalog registration failed. The plan name \"%s\" is a reserved name; you must choose a different plan name.

File: pkg/util/errors.go [hunk 2]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
 	case ErrReferencedInstanceNotShared:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, must be shared first.", entityName),
+			Description: fmt.Sprintf("Failed to create the instance. The referebced instance %s, must be shared first.", entityName),
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:272]: Failed to create the instance. The instance %s, to which you want to refer, must be shared first.

File: pkg/util/errors.go [hunk 3]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrReferencedInstanceNotFound:
 		return &HTTPError{
 			ErrorType:   "NotFound",
-			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, not found.", entityName),
+			Description: fmt.Sprintf("Failed to create the instance. The referenced instance %s, not found.", entityName),
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:278]: Failer to create the instance. The instance %s, to which you want to refer, was not found.

File: pkg/util/errors.go [hunk 4]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 			StatusCode:  http.StatusNotFound,
 		}
 	case ErrReferenceWithWrongServiceOffering:
 		return &HTTPError{
 			ErrorType:   "NotFound",
-			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, belongs to a different service.", entityName),
+			Description: fmt.Sprintf("Failed to create the instance. The referenced instance %s, not found for this service offering.", entityName),
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:284]: Failed to create the instance. The instance %s, to which you want to refer, doesn't belong to the specified service offering.

File: pkg/util/errors.go [hunk 5]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 			StatusCode:  http.StatusNotFound,
 		}
 	case ErrMultipleReferenceSelectorResults:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: "Failed to create the reference. There are multiple shared instances that match your criteria. Specify the ID of the instance to which you want to refer.",
-			StatusCode:  http.StatusBadRequest,
+			Description: "Failed to create the instance. There are multiple shared instances that match your criteria. Refine the search criteria or specify an instance ID.",
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:290]: No change.

File: pkg/util/errors.go [hunk 6]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
+			StatusCode:  http.StatusNotFound,
 		}
 	case ErrInvalidReferenceSelectors:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: "Failed to create the reference. Your query selectors are invalid.",
+			Description: "Failed to create the instance. Your search parameters are invalid.",
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:296]: No change.

File: pkg/util/errors.go [hunk 7]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrNoResultsForReferenceSelector:
 		return &HTTPError{
 			ErrorType:   "NotFound",
-			Description: "Failed to create the reference. Couldn't find a shared instance that matches your criteria.",
+			Description: "Failed to create the instance. Could not find a shared instance that matches your criteria. Refine the search criteria or specify an instance ID.",
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:302]: Failed to create the instance. Could not find a shared instance that matches your criteria. Refine your search criteria or specify the instance ID.

File: pkg/util/errors.go [hunk 8]
```diff
@@ -270,43 +269,43 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 			StatusCode:  http.StatusNotFound,
 		}
 	case ErrChangingPlanOfReferenceInstance:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Failed to update the instance %s. It isn't allowed to change the plan of reference instances.", entityName),
+			Description: fmt.Sprintf("Failed to update the instance %s. It isn't allowed to change the plan of reference instances.", entityName), // updateable false. we should use the same error message.
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:308]: Failed to update the instance %s. It is not allowed to change the plan of a reference instance.

File: pkg/util/errors.go [hunk 9]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
 	case ErrChangingParametersOfReferenceInstance:
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Failed to update the instance %s. It isn't allowed to change a reference instance.", entityName),
+			Description: fmt.Sprintf("Failed to update the instance %s. It isn't allowed to change the parameters of reference instance.", entityName),
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:326]: Failed to update the instance %s. It is not allowed to change the parameters of a reference instance.

File: pkg/util/errors.go [hunk 10]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
-	case ErrUnknownOSBMethod:
-		return &HTTPError{
-			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Unknown OSB method: \"%s\".", entityName),
+			Description: "Failed to create the instance. Specify an instance ID or use search criteria.", // consider to mention the help.sap documentation on the plan's description.
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrSharedPlanHasReferences:
-		errorMessage := fmt.Sprintf("Couldn't update the service plan. Before you can set it as %s=false, you first need to unshare the instances of the plan: %s.", instance_sharing.SupportsInstanceSharingKey, entityName)
+		errorMessage := fmt.Sprintf("Could not update the service plan %s. Before you can set it as %s=false, you first need to unshare its shared instances.", instance_sharing.SupportsInstanceSharingKey, entityName)
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:336]: "Failed to update the service plan %s. Before you can set %s=false, you first need to unshare its shared instances."

also, change the order of the strings!

File: pkg/util/errors.go [hunk 11]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 		return &HTTPError{
 			ErrorType:   "BadRequest",
 			Description: errorMessage,
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrInvalidShareRequest:
-		errorMessage := fmt.Sprintf("Could not update the instance %s. Modifying the \"shared\" property should be in a dedicated request that doesn't contain any other properties.", entityName)
+		errorMessage := fmt.Sprintf("Could not update the instance %s. It is not allowed to modify the \"shared\" property on the same request with any other properties.", entityName)
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:343]: Failed to update the instance %s. It isn't allowed to modify the "shared" property in the request that modifies also other properties.

File: pkg/util/errors.go [hunk 12]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 		return &HTTPError{
 			ErrorType:   "BadRequest",
 			Description: errorMessage,
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrInvalidProvisionRequestWithSharedProperty:
-		errorMessage := "Failed provisioning the instance. The \"shared\" property should be in a dedicated request that doesn't contain any other properties."
+		errorMessage := "Failed provisioning the instance. Setting the instance as \"shared\" is only possible after the instance is created."
```
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:350]: Failed to create the instance. Setting the instance as \"shared\" is only possible after the instance has been created.

File: pkg/util/errors.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (EpsilonTal) [reply pkg/util/errors.go:336]: "Failed to update the service plan %s. Before you can set %s to false, you first need to unshare its shared instances."

**Review by idobergerfroind — DISMISSED** [review]

File: pkg/util/errors.go [hunk 13]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
 		return &HTTPError{
 			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Failed to create the instance. Missing or invalid parameter \"%s\".", entityName),
-			StatusCode:  http.StatusBadRequest,
-		}
-	case ErrUnknownOSBMethod:
-		return &HTTPError{
-			ErrorType:   "BadRequest",
-			Description: fmt.Sprintf("Unknown OSB method: \"%s\".", entityName),
+			Description: "Failed to create the instance. Specify an instance ID or use search criteria.", // consider to mention the help.sap documentation on the plan's description.
```
Comment (i522006) [inline pkg/util/errors.go:332]: Just this one message still seems strange to me.
Why did it fail?
"Specify the instance ID or use search criteria?"
Something here doesn't seem right.

File: pkg/util/errors.go [hunk 14]
```diff
@@ -324,44 +323,38 @@ func HandleInstanceSharingError(err error, entityName string) error {
...
-		errorMessage := fmt.Sprintf("Couldn't update the service plan. Before you can set it as %s=false, you first need to unshare the instances of the plan: %s.", instance_sharing.SupportsInstanceSharingKey, entityName)
+		errorMessage := fmt.Sprintf("Failed to update the service plan %s. Before you can set %s to false, you first need to unshare its shared instances.", entityName, instance_sharing.SupportsInstanceSharingKey)
 		return &HTTPError{
 			ErrorType:   "BadRequest",
 			Description: errorMessage,
 			StatusCode:  http.StatusBadRequest,
 		}
 	case ErrInvalidShareRequest:
-		errorMessage := fmt.Sprintf("Could not update the instance %s. Modifying the \"shared\" property should be in a dedicated request that doesn't contain any other properties.", entityName)
+		errorMessage := fmt.Sprintf("Failed to update the instance %s. It is not allowed to modify the \"shared\" property in the request that modifies also other properties.", entityName)
```
Comment (i522006) [inline pkg/util/errors.go:343]: that also modifies other properties.

**Review by idobergerfroind — APPROVED** [review]

**Pull request merged**